chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -780,22 +783,32 @@ func (b *BitcoindNotifier) historicalSpendDetails(
 	spendRequest chainntnfs.SpendRequest, startHeight, endHeight uint32) (
 	*chainntnfs.SpendDetail, error) {
 
-	// Begin scanning blocks at every height to determine if the outpoint
-	// was spent.
+	// batchrequest will store the scan requests at every height to determine
+	// if the output was spent
+	batchRequest := []rpcclient.FutureGetBlockHashResult{}
 	for height := endHeight; height >= startHeight && height > 0; height-- {
-		// Ensure we haven't been requested to shut down before
-		// processing the next height.
+		// BatchClient works with AsynRequest
+		sendReq, err := b.chainConn.GetBlockHashAsync(int64(height))
+		if err != nil {
+			return nil, fmt.Errorf("unable to receive the block promise %v", err)
+		}
+		batchRequest = append(batchRequest, sendReq)
+	}
+	// Sending the bulk request using updated batchClient
+	b.chainConn.Send()
+
+	for height, response := range batchRequest {
```

Comment:
With the slice this isn't really the `height` anymore but the index, which requires the change in the error message below. With the map (see comment above), this would become the `height` again.

Reply:
Updated!